internal/bloblang/query/methods_strings.go

```diff
@@ -1940,3 +1940,63 @@ root.description = this.description.trim()`,
 		}, nil
 	},
 )
+
+var _ = registerSimpleMethod(
+	NewMethodSpec(
+		"trim_prefix", "",
+	).InCategory(
+		MethodCategoryStrings,
+		"Remove the provided leading prefix substring from a string. If the string does not have the prefix substring, it is returned unchanged.",
+		NewExampleSpec("",
+			`root.name = this.name.trim_prefix("foobar_")
+root.description = this.description.trim_prefix("foobar_")`,
+			`{"description":"unchanged","name":"foobar_blobton"}`,
+			`{"description":"unchanged","name":"blobton"}`,
+		),
+	).Param(ParamString("prefix", "The leading prefix substring to trim from the string.")),
+	func(args *ParsedParams) (simpleMethod, error) {
+		prefix, err := args.FieldString("prefix")
+		if err != nil {
+			return nil, err
+		}
```

Comment:
You could add byte conversion here and avoid the conversion each execution:
```
bytesPrefix = []byte(prefix)
```

Reply:
Good point, thanks! ❤️